### PR TITLE
CURATOR-232 Consolidate Timing and Client

### PR DESCRIPTION
**Pull request by madrob**
Pull references to Timing and CuratorFramework client into BeforeMethod
and AfterMethod invocations so that we can manage them more efficiently.

### CODE REVIEW

File: curator-framework/src/test/java/org/apache/curator/framework/FrameworkTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/*
```
Comment (Randgalt) [inline curator-framework/src/test/java/org/apache/curator/framework/FrameworkTest.java]: I don't see enough value in this class to maintain it. In order for it to be used in the recipes module we'd have to build a test jar and include and/or create another test module. Create new clients isn't always consistent in the tests and it's only 1 line of code anyway. The closeable stuff could be pushed into BaseClassForTests by having a List<Closeable> that can be added to and auto closed on @After.

**Comment by Randgalt** [comment]
I'm concerned about the number of changes in this PR. I like the idea of putting some common functionality in BaseClassForTests but I think it's too risky to touch all the tests. I suggest that we add the things to BaseClassForTests but leave the tests as is an updated them as needed over time.

File: curator-test/src/main/java/org/apache/curator/test/BaseClassForTests.java [hunk 2]
```diff
@@ -55,6 +63,8 @@
         INTERNAL_RETRY_FAILED_TESTS = retryFailedTests;
     }
 
+    private List<Closeable> toClose;
```
Comment (Randgalt) [inline curator-test/src/main/java/org/apache/curator/test/BaseClassForTests.java]: Is this needed anymore given that Java 7 has Autocloseable?

**Comment by tisonkun** [comment]
Closed as stale. There're many changes since this patch was made in the first place. Feel free to resubmit it if it's still relevant.

**Pull request closed without merging**